core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -532,7 +532,8 @@ public PersistenceState getPersistenceState(long fileId) throws FileDoesNotExist
       Inode<?> inode = null;
       if (inodePath.fullPathExists()) {
         inode = inodePath.getInode();
-        if (inode.isDirectory() && ((InodeDirectory) inode).isDirectChildrenLoaded()) {
+        if ((inode.isDirectory() && ((InodeDirectory) inode).isDirectChildrenLoaded()) || inode
+            .isFile()) {
```

Comment:
Hi @gpang  I think I remember it wrong. 1.0.1 didn't do this.  Ignore my comment then.   

 public long getFileId(AlluxioURI path) throws AccessControlException {
    synchronized (mInodeTree) {
      Inode inode;
      try {
        checkPermission(FileSystemAction.READ, path, false);
        inode = mInodeTree.getInodeByPath(path);
      } catch (InvalidPathException e) {
        try {
          return loadMetadata(path, true);
        } catch (Exception e2) {
          return IdUtils.INVALID_FILE_ID;
        }
      }
      return inode.getId();
    }
  }